src/windows/osd.c

```diff
@@ -647,3 +647,32 @@ struct ofi_pollfds_ctx *ofi_pollfds_alloc_ctx(struct ofi_pollfds *pfds, int fd)
 	ctx->index = pfds->nfds++;
 	return ctx;
 }
+
+char *strcasestr(const char *haystack, const char *needle)
+{
+    char *uneedle, *uhaystack, *pos = NULL;
+    int i;
+
+    uneedle = malloc(strlen(needle) + 1);
+    uhaystack = malloc(strlen(haystack) + 1);
+    if (!uneedle || !uhaystack)
+        goto out;
+
+    for (i = 0; i < strlen(needle); i++)
+        uneedle[i] = (char) toupper(needle[i]);
+    uneedle[i] = '\0';
+
+    for (i = 0; i < strlen(haystack); i++)
+        uhaystack[i] = (char) toupper(haystack[i]);
+    uhaystack[i] = '\0';
+
+    pos = strstr(uhaystack, uneedle);
+    if (pos)
+        pos = (char *) ((uintptr_t) haystack + (uintptr_t) pos -
+                (uintptr_t) uhaystack);
+out:
+    free(uneedle);
+    free(uhaystack);
+    return pos;
+}
```

Comment:
Your editor isn't set correctly.  These should use tabs, not spaces, for indentation.  Tabs should show as 8 characters to ensure alignment.